integration/test/multireleasehandling/multi_release_handling_test.go

```diff
@@ -170,7 +192,75 @@ func TestMultiReleaseHandling(t *testing.T) {
 
 			return nil
 		}
-		b := backoff.NewMaxRetries(30, 5*time.Second)
+		b := backoff.NewMaxRetries(35, 6*time.Second)
+
+		err := backoff.Retry(o, b)
+		if err != nil {
+			t.Fatalf("err == %v, want %v", err, nil)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checked App CRs created for Release CRs components"))
+	}
+
+	// Create the ReleaseCycle for the Release CR 1 marking it as EOL release.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating ReleaseCycle CR %#q", releaseCycleCR1.Name))
+
+		_, err := config.K8sClients.G8sClient().ReleaseV1alpha1().ReleaseCycles().Create(releaseCycleCR1)
+		if err != nil {
+			t.Fatalf("err == %v, want %v", err, nil)
+		}
+
+		defer func() {
+			if env.CircleCI() || env.KeepResources() {
+				return
+			}
+
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("cleaning up ReleaseCycle CR %#q", releaseCycleCR1.Name))
+
+			err := config.K8sClients.G8sClient().ReleaseV1alpha1().ReleaseCycles().Delete(releaseCycleCR1.Name, &metav1.DeleteOptions{})
+			if err != nil {
+				config.Logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("failed to clean up ReleaseCycle CR %#q", releaseCycleCR1.Name), "stack", fmt.Sprintf("%#v", err))
+			}
+
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("cleaned up ReleaseCycle CR %#q", releaseCycleCR1.Name))
+		}()
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created ReleaseCycle CR %#q", releaseCycleCR1.Name))
+	}
+
+	// Check if App CRs for all components of Release CR 1 not shared with
+	// the Release CR 2 are deleted.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking App CRs created for Release CRs components"))
+
+		o := func() error {
+			list, err := config.K8sClients.G8sClient().ApplicationV1alpha1().Apps("").List(metav1.ListOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			var appCRNames []string
+			for _, obj := range list.Items {
+				appCRNames = append(appCRNames, obj.Name)
+			}
+
+			expectedAppCRNames := []string{
+				"aws-operator.4.7.0",
+				"cert-operator.0.1.0",
+				"release-aws.v6.1.0",
+			}
+
+			sort.Strings(appCRNames)
+			sort.Strings(expectedAppCRNames)
+
+			if !cmp.Equal(appCRNames, expectedAppCRNames) {
+				return microerror.Maskf(waitError, "\n\n%s\nappCRNames = %#v\nexpectedAppCRNames = %#v\n\n", cmp.Diff(appCRNames, expectedAppCRNames), appCRNames, expectedAppCRNames)
+			}
+
+			return nil
+		}
+		b := backoff.NewMaxRetries(65, 6*time.Second)
```

Comment:
I increased that so it's enough time to wait for a second loop to complete on the CI machine. ResyncPeriod is 2m.